crates/turborepo-lib/src/run/scope/target_selector.rs

```diff
@@ -141,26 +131,34 @@ impl FromStr for TargetSelector {
             };
 
             // strip the square brackets
-            let inner_str = commits_str
+            let commits_str = commits_str
                 .strip_prefix('[')
-                .and_then(|s| s.strip_suffix(']'));
+                .and_then(|s| s.strip_suffix(']'))
+                .expect("regex guarantees square brackets");
+            if commits_str.is_empty() {
+                return Err(InvalidSelectorError::InvalidGitRange(
+                    commits_str.to_string(),
+                ));
+            }
 
-            if let Some(commits_str) = inner_str {
-                if let Some((a, b)) = commits_str.split_once("...") {
-                    (a.to_string(), b.to_string())
-                } else {
-                    (commits_str.to_string(), String::new())
+            let git_range = if let Some((a, b)) = commits_str.split_once("...") {
+                GitRange {
+                    from_ref: a.to_string(),
+                    to_ref: Some(b.to_string()),
```

Comment:
Do we need to check if `b` is empty? If I'm parsing the regex correctly I think it can be empty.

Reply:
Oh yeah, good idea. The validation story here has not been great...